packages/web-components/fast-foundation/src/number-field/number-field.ts

```diff
@@ -158,28 +177,47 @@ export class NumberField extends FormAssociatedNumberField {
     public control: HTMLInputElement;
 
     /**
-     *
+     * Gate that prevents valueChanged from calling itself while updating this.value
+     * @internal
+     */
+    private updatingValue: boolean = false;
+
+    /**
+     * Validates that the value is a number between the min and max
      * @param previousValue - previous stored value
      * @param nextValue - value being updated
+     * @internal
      */
     public valueChanged(previousValue, nextValue): void {
-        let value: number | string = parseFloat(nextValue);
-        if (isNaN(value)) {
-            value = "";
-        } else {
-            value = Math.min(value, this.max ?? value);
-            value = Math.max(value, this.min ?? value);
-        }
-
-        this.value = value.toString();
-
-        if (this.proxy instanceof HTMLInputElement) {
-            this.proxy.value = this.value;
-        }
-
-        if (previousValue !== undefined) {
-            this.$emit("input");
-            this.$emit("change");
+        if (previousValue != nextValue && this.updatingValue === false) {
+            this.updatingValue = true;
+            let value: number | string = parseFloat(nextValue);
+            if (isNaN(value)) {
+                value = "";
+            } else {
+                value = Math.min(value, this.max ?? value);
+                value = Math.max(value, this.min ?? value).toString();
+            }
+
+            if (this.proxy instanceof HTMLInputElement) {
+                this.proxy.value = value;
+            }
+
+            if (
+                this.control &&
+                (this.control.value != nextValue || previousValue === null)
+            ) {
+                this.control.value = value;
+            }
+
+            this.value = value;
+
+            if (previousValue !== undefined) {
+                this.$emit("input");
+                this.$emit("change");
+            }
+            this.updatingValue = false;
+            super.valueChanged(previousValue, nextValue);
```

Comment:
Should the `$emit` check happen after calling the supermethod?

Reply:
It could happen after the calling super. In the case that they are submitting form on change maybe? The only thing that happens in the super is setting the proxy which is only for form submission. I'll move the emit.

Reply:
I moved the emits to the end of valueChanged().